go/vt/client/fakeserver_test.go

```diff
@@ -0,0 +1,233 @@
+package client
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+
+	mproto "github.com/youtube/vitess/go/mysql/proto"
+	"github.com/youtube/vitess/go/sqltypes"
+	"github.com/youtube/vitess/go/vt/topo"
+	"github.com/youtube/vitess/go/vt/vtgate/proto"
+	"github.com/youtube/vitess/go/vt/vtgate/vtgateservice"
+	"golang.org/x/net/context"
+)
+
+// fakeVTGateService has the server side of this fake
+type fakeVTGateService struct {
```

Comment:
We already have a fakeVTGateService, what's the point to not reuse it ? https://github.com/youtube/vitess/blob/master/go/vt/vtgate/vtgateconntest/client.go
In the meanwhile, I'd like to bring up a question that maybe we could consolidate our fake structs so that people could easily reuse existing ones.